cmd/zed/zed_file.c

```diff
@@ -167,17 +168,23 @@ zed_file_is_locked(int fd)
 void
 zed_file_close_from(int lowfd)
 {
-	const int maxfd_def = 256;
-	int errno_bak;
+	static const int maxfd_def = 256;
+	int errno_bak = errno;
 	struct rlimit rl;
-	int maxfd;
+	int maxfd = 0;
 	int fd;
-
-	errno_bak = errno;
-
-	if (getrlimit(RLIMIT_NOFILE, &rl) < 0) {
-		maxfd = maxfd_def;
-	} else if (rl.rlim_max == RLIM_INFINITY) {
+	DIR *fddir;
+	struct dirent *fdent;
+
+	if ((fddir = opendir("/proc/self/fd")) != NULL) {
```

Comment:
Nice improvement in `zed_file_close_from()`.  My only concern is this optimization is Linux specific.  That's not currently a problem since FreeBSD does not use the ZED, but it's my understanding the work to add MacOS support will use it.  However, I suppose that's probably fine to for the moment since worst case it'll fallback to the existing logic.

Reply:
I optimised with the thought that this is only used on Linux, that's true, but libbsd's closefrom() implementation that I based the approach on just uses the same code but with a different path, trivial fix.